src/app/dev.js

```diff
@@ -14,90 +15,89 @@ import { getModuleContext, getChanged } from './util/webpack';
  * Register all templates beforehand with a require context, and save the partial by filename
  */
 const [partialModules, partialsContext] = getModuleContext(
-	require.context('./component/blocks/', true, /\.hbs$/),
-	(context, key, module) => {
-		Handlebars.registerPartial(/\/([^/]+)\.hbs/gi.exec(key)[1], module)
-	}
+  require.context('./component/blocks/', true, /\.hbs$/),
+  (context, key, module) => {
+    Handlebars.registerPartial(/\/([^/]+)\.hbs/gi.exec(key)[1], module);
+  },
 );
 
 /**
  * json data context, for hot reloading
  */
-const [jsonModules, jsonContext] = getModuleContext(
-	require.context('../data/', true, /\.json$/),
-);
+const [jsonModules, jsonContext] = getModuleContext(require.context('../data/', true, /\.json$/));
 
 // Get info for current page
 const pageMatch = /\/(.*)\.html/i.exec(document.location.pathname);
-const jsonModuleName = (pageMatch && pageMatch[1] || 'index');
-const getJsonData = () => {
-	return jsonModules[`./${jsonModuleName}.json`];
-};
-
+const jsonModuleName = (pageMatch && pageMatch[1]) || 'index';
+const getJsonData = () => jsonModules[`./${jsonModuleName}.json`];
 
 function render() {
-	const div = document.getElementById('app');
+  const div = document.getElementById('app');
 
-	if (jsonModuleName === 'index') {
-		// render the index overview page
-		div.innerHTML = indexTemplate({
-			pages: Object.keys(jsonModules)
-				.map(key => ({
-					page: key.split('/').pop().split('.').slice(0, -1).join('.'),
-					data: jsonModules[key],
-				}))
-				.sort()
-				.map(({ page, data }) => ({
-					page,
-					link: page + '.html',
-					data,
-				}))
-		});
-	} else {
-		// render page with data
-		div.innerHTML = appTemplate(getJsonData());
+  if (jsonModuleName === 'index') {
+    // render the index overview page
+    div.innerHTML = indexTemplate({
+      pages: Object.keys(jsonModules)
+        .map(key => ({
+          page: key
+            .split('/')
```

Comment:
Don't know the exact value of key but isn't this solvable by using a regex. If you're doing path based things, you could also have a look at `path` module if there is something that might fit your needs.

Reply:
Didn't think of using `path.basename` before because this is not a nodejs script, but using webpack (and this file is only used in dev) it works fine :)